packages/core/src/error.ts

```diff
@@ -13,14 +15,45 @@ export class ApplicationError extends Error {
   constructor(
     public readonly type: string,
     public readonly status: number,
-    public readonly title: string
+    public readonly title: string,
+    public readonly detail?: string
   ) {
-    super(title)
+    super(detail || title)
```

Comment:
Use nullish coalesce
```suggestion
    super(detail ?? title)
```

Reply:
Agreed 👍